database_normalization/database_normalization.md

```diff
@@ -0,0 +1,370 @@
+
+Whether these tables of interactions have primary keys and their own identifiers depends, but they often will.  For example, a "device\_shipment" table might have a primary key called "device\_shipment\_id".
+
+### Why Bother With Normalization?
+
+To show why organizing data in a normalized way is helpful, let's use an example.
+
+Which of the following is more efficient to correct, if we want to change "orane juice" to "orange juice"? Is it easier to correct with the data stored in one table, or with the data in two tables, where items are given an id number?
+
+**One Table Option**
+
+**orders**
+
+<!-- data-type="none" class="tight-table" -->
+| order_num   | item_1   | item_2  | ...  | item_50  |
+| :--------- | :--------- | :--------- | :--------- | :--------- |
+| 23125    | orane juice     | pistachios     | ...    | `NULL`    |
+| 41320    | peanut butter    | plain bagels     | ...    | orane juice    |
+| 53011    | napkins    | distilled water     | ...    | `NULL`    |
+| 14123    | pistachios    | orane juice     | ...    | peanut butter    |
+
+In the one table option, we have to look for the phrase "orane juice" anywhere in 50 columns across all rows (consider that there could be thousands of rows).
+
+**Two Table Option**
+
+**items**
+
+*Note: here we are using the "item\_id" field as a primary key as well, which is why we include (PK) in the table header.  Often, database tools will do something similar and display a special symbol (like <span class="fa-solid fa-key"></span>) or the letters "PK" to draw your attention to the fact that a particular field is a primary key.* 
+
+<!-- data-type="none" class="tight-table" -->
+| item_id (PK)  | item_name  |
+| :--------- | :--------- |
+| 15 | distilled water |
+| 178 | napkins |
+| 210 | orane juice |
+| 97 | peanut butter |
+| 108 | pistachios  |
+| 233 | plain bagels  |
+
+**order_items**
+
+*Note: here we want to allow the "order\_id" to repeat as many times as necessary to include all items, so it's not a primary key.* 
+
+<!-- data-type="none" class="tight-table" -->
+| order_id | item_id |
+| :--------- | :--------- |
+| 23125 | 210 |
+| 23125 | 108 |
+| 41320 | 97 |
+| 41320 | 233 |
+| 41320 | 210  |
+| 53011 | 178 |
+| 53011 | 15 |
+| 14123 | 108 |
+| 14123 | 210 |
+| 14123 | 97 |
+
+In the two table option, we only have to find "orane juice" in a single column, "item\_name", and it occurs once only (this is because this id number is a primary key, which means it can only occur once).
```

Comment:
```suggestion
In the two table option, we only have to find "orane juice" in a single column, "item\_name", and it occurs once only (this is because this ID number is a primary key, which means it can only occur once).
```